Sources/AsyncHTTPClient/HTTPClient+HTTPCookie.swift

```diff
@@ -124,51 +94,116 @@ extension HTTPClient {
         ///     - value: The cookie's string value.
         ///     - path: The cookie's path.
         ///     - domain: The domain of the cookie, defaults to nil.
-        ///     - expires: The cookie's expiration date, defaults to nil.
+        ///     - expires_timestamp: The cookie's expiration date, as a number of seconds since the Unix epoch. defaults to nil.
         ///     - maxAge: The cookie's age in seconds, defaults to nil.
         ///     - httpOnly: Whether this cookie should be used by HTTP servers only, defaults to false.
         ///     - secure: Whether this cookie should only be sent using secure channels, defaults to false.
-        public init(name: String, value: String, path: String = "/", domain: String? = nil, expires: Date? = nil, maxAge: Int? = nil, httpOnly: Bool = false, secure: Bool = false) {
+        internal init(name: String, value: String, path: String = "/", domain: String? = nil, expires_timestamp: Int64? = nil, maxAge: Int? = nil, httpOnly: Bool = false, secure: Bool = false) {
             self.name = name
             self.value = value
             self.path = path
             self.domain = domain
-            self.expires = expires
+            self.expires_timestamp = expires_timestamp
             self.maxAge = maxAge
             self.httpOnly = httpOnly
             self.secure = secure
         }
+    }
+}
 
-        func parseComponent(_ component: String) -> (String?, String?) {
-            let nameAndValue = component.split(separator: "=", maxSplits: 1).map {
-                $0.trimmingCharacters(in: .whitespaces)
-            }
-            if nameAndValue.count == 2 {
-                return (nameAndValue[0].lowercased(), nameAndValue[1])
-            } else if nameAndValue.count == 1 {
-                return (nameAndValue[0].lowercased(), nil)
-            }
-            return (nil, nil)
-        }
+extension HTTPClient.Response {
+    /// List of HTTP cookies returned by the server.
+    public var cookies: [HTTPClient.Cookie] {
+        return self.headers["set-cookie"].compactMap { HTTPClient.Cookie(header: $0, defaultDomain: self.host) }
     }
 }
 
 extension String {
-    fileprivate func omittingQuotes() -> String {
-        let dquote = "\""
-        if !hasPrefix(dquote) || !hasSuffix(dquote) {
-            return self
+    fileprivate init(utf8Slice: String.UTF8View.SubSequence, in base: String) {
+        self.init(base[utf8Slice.startIndex..<utf8Slice.endIndex])
```

Comment:
Why is it necessary for us to slice the base? Isn't `utf8Slice` already the subsequence we're passing here?

Reply:
- `String.init?(String.UTF8View.Subsequence)` can fail if the indexes are not aligned to character boundaries. We don't expect that to happen, but the SSWG guidelines also say to avoid force-unwrapping wherever possible.

- `String.init(Substring)` (used here) can not fail. If the indexes are not aligned to character boundaries, it will adjust them. It's... not unreasonable fallback behaviour, and means we can avoid a bunch of optionals that won't be ever be nil.

In terms of performance, they both allocate independent String values, so are both O(n). I would suppose the `String.init(Substring)` initializer can be slightly faster, as by widening the bounds to aligned character boundaries, it may be able to skip UTF-8 validation.

Reply:
I've added a doc-comment, explaining how this is different from `String.init?(String.UTF8View.Subsequence)`

Reply:
I think I'd rather have the optionals, tbh. The SSWG guidelines are understandable, but we're asserting that these are ASCII strings, and so there can never be a case where we have alignment issues. However, I'd like to hear from @fabianfett (SSWG member with a lot of interest in AHC) and @tomerd (SSWG member) for some more guidance here.

Reply:
Well, technically we're only asserting that our processing (trimming, etc) operates on whole scalars, which is fine since we only do ASCII-level processing.

Force-unwrapping exists for occasions where we have an optional which we can determine will never be nil, which IMO is covered by this. But yeah, the SSWG guidelines say we should avoid that, and there is a relatively easy way around it by going through `Substring`. I don't mind either way.